Docs pm 2.9 dh (#143)
mulesoft/docs-partner-manager: modules/ROOT/pages/create-outbound-message-flow.adoc

. Click *New message flow > Send to partner*.
. Either select an existing partner or select *New* to create a new partner, as described in xref:create-partner.adoc#[Create a partner].
. Configure the receiving endpoint.
. Configure the source message type.
. Configure reference IDs for the partner and host.
. Import the translation map.
. Configure the target message type.
. Configure the target endpoint.
. Verify that the message flow is complete.

[[source-at-host-endpoint]]
== Configure the Source Endpoint

The source endpoint in an outbound message flow receives application messages from the backend applications. Partner Manager supports the FTP, HTTP, HTTPS, and SFTP formats for the source endpoint.

To configure the source endpoint:

. In the message flow, expand the *Source at <host-name>* section and click *Select*.
. In the *Select an endpoint: Source at <host-name>* window, do either of the following:
* Select an existing source at host endpoint and click *Select*.
* Click *New* to create a new source at host endpoint, as described in xref:create-endpoint.adoc[Create and Configure Endpoints].

[[source-message-type]]
== Configure the Source Message Type

> The source message type in an outbound message flow defines the structure of the messages that Partner Manager sends to a partner, either directly or through a third party connection. Partner Manager supports the JSON and XML source message types.

Reviewer: Please rephrase as below:

The source message type in an outbound message flow defines the structure of the messages that Partner Manager receives from the backend applications, to be translated to a configured send to partners message type, and sent to your partners either directly or through a third party connection. Partner Manager supports the JSON and XML source message types in outbound message flows.

* Select an existing source at host endpoint and click *Select*.
* Click *New* to create a new source at host endpoint, as described in xref:create-endpoint.adoc[Create and Configure Endpoints].

[[source-message-type]]
== Configure the Source Message Type

The source message type in an outbound message flow defines the structure of the messages that Partner Manager receives from the backend applications. Partner Manager translates these messages to a configured send to partners message type, and then sends them your partners, either directly or through a third party connection. 

Partner Manager supports the JSON and XML source message types in outbound message flows.

To create the source message type:

. In the message flow, expand the *Source message type* section and click *Select*.
. In the *Source message type* window, do either of the following:
* Select an existing source message type and click *Select*.
* Click *New* to create a new source message type, as described in xref:partner-manager-create-message-type[Create and Configure Message Types].

[[reference-ids]]
== Configure Reference IDs for the Partner and Host

> Partner Manager uses the partner reference ID and host reference ID to locate the appropriate message flow for routing outbound messages. For information about about how this routing process works, see xref:partner-manager-identifiers.adoc[Sender and Receiver Identifiers] and xref:outbound-message-routing.adoc[Outbound Message Routing].

Reviewer 2: ```suggestion
Partner Manager uses the partner reference ID and host reference ID to locate the appropriate message flow for routing outbound messages. For information about how this routing process works, see xref:partner-manager-identifiers.adoc[Sender and Receiver Identifiers] and xref:outbound-message-routing.adoc[Outbound Message Routing].
```

To configure a partner reference ID, either select *All available identifiers* or select a specific partner identifier.

To configure a host reference ID, either select *Ignore* or select your organization's internal reference ID (such as the business unit name) as it appears in the data element of the source JSON or XML message from your backend applications.

[[translation-map]]
== Import the Translation Map

> A translation map in an outbound message flow transforms outbound messages from your organization's enterprise application format to the format your partners expect. Create a translation map by using DataWeave. For more information, see see xref:partner-manager-maps.adoc[Translation Maps].

Reviewer 2: ```suggestion
A translation map in an outbound message flow transforms outbound messages from your organization's enterprise application format to the format your partners expect. Create a translation map by using DataWeave. For more information, see xref:partner-manager-maps.adoc[Translation Maps].
```

. In the *<protocol> Identifiers* section, select the sender and receiver identifiers to use on the X12 ISA and GS segments or the EDIFACT UNB segments.
+
These identifiers are configured in the host and partner profiles.

[[send-to-partner-endpoint]]
== Configure the Send To Partner Endpoint

The send to partner endpoint is the target endpoint to which transformed messages are sent to partners, either directly or via third party connections. Use a third-party-owned send to partner endpoint for partners that use a third party connection.

Partner Manager supports the EDIFACT and X12 protocols for send to partner endpoints.

To create a send to partner endpoint:

. In the message flow, expand the *Send to <partner>* section and click *Select*.
. Do either of the following:
* Select an existing send to endpoint and click *Select*.
+
You can select any previously configured Send to partner endpoints owned by the host, the partner for whom the message flow is configured, or a third party connection.
+
* Click *New* to create a new send to endpoint, as described in xref:create-endpoint.adoc[Create and Configure Endpoints].

> If you selected AS2 as the protocol and there is no certificate associated with the partner or third party profile, import the certificate of the partner or third-party connection that owns the endpoint.

Reviewer 2: ```suggestion
If you selected AS2 as the protocol and there is no certificate associated with the partner or third party profile, import the certificate of the partner or third party connection that owns the endpoint.
```

= Create Outbound Message Flows

> Create outbound message flows to receive application messages from your backend systems, validate and transform these messages to the format your partners expect, and then send them to your partners either directly, or through 3rd party connections.

Reviewer 3: ```suggestion
Create outbound message flows to receive application messages from your backend systems, validate and transform these messages to the format your partners expect, and then send them to your partners, either directly or through third-party connections.
```

> Before you create your first message flow, configure your xref:setup-payload-storage-API.adoc[payload storage API].

Reviewer 3: This is conveyed by way of a Before You Begin topic in the inbound message flow task, isn't it?
Either is fine; I kind of prefer this treatment when the prereq is just one thing....

To create an outbound message flow:

. From the Anypoint Partner Manager navigation menu, select *Message Flows*.
. Click *New message flow > Send to partner*.
. Either select an existing partner or select *New* to create a new partner, as described in xref:create-partner.adoc#[Create a partner].
+

> A new message flow is created.

Reviewer 3: Dunno' if we really need this.

Reviewer 3: I still think it's weird to make the configuration part of the creation without specifying that in the title....
Would it work to end the create task here and then follow with a sentence and a new list of links:

```suggestion
After you create a new message flow, customize it by performing the following tasks:
```
number list or steps, as you think most appropriate

+

> . <<source-at-host-endpoint,Configure the receiving endpoint>>.

Reviewer 3: ```suggestion
. <<source-at-host-endpoint,Configure the source endpoint>>.
```

[[reference-ids]]
== Configure Reference IDs for the Partner and Host

Partner Manager uses the partner reference ID and host reference ID to locate the appropriate message flow for routing outbound messages. For information about about how this routing process works, see xref:partner-manager-identifiers[Sender and Receiver Identifiers] and xref:outbound-message-routing.adoc[Outbound Message Routing].

To configure a partner reference ID, either select *All available identifiers* or select a specific partner identifier.

To configure a host reference ID, either select *Ignore* or select your organization's internal reference ID (such as the business unit name) as it appears in the data element of the source JSON or XML message from your backend systems.

[[translation-map]]
== Import the Translation Map

A translation map in an outbound message flow transforms outbound messages from your organization's enterprise application format to the format your partners expect. Create a translation map by using DataWeave. For more information, see see xref:partner-manager-maps.adoc[EDI Translation Maps].

To import the translation map:

. In the message flow, expand the *Map* section.

> . Click *Import < Choose file*.

Reviewer 3: ```suggestion
. Click *Import > Choose file*.
```

. Navigate to the DataWeave map you want to use and click *Open*.
. Click *Upload mapping*.

> The translation map is added to the message flow.

Reviewer 3: ditto prev. ? about the need to confirm success

[[send-to-partner-message-type]]
== Configure the Target Message Type

> The target message type in an outbound message flow defines the structure of the messages that Partner Manager sends to the partner, either directly or via a 3rd party connection. Partner Manager supports the AS2, FTP, and SFTP formats for send to message types.

Reviewer 3: "send to message types" = "target message types"?
If you STET, then hyphenate: "send-to message types" or "send-to-message types," as appropriate.

For the EDIFACT and X12 message types, configuring the target message type also involves configuring acknowledgements (optional), creating custom EDIFACT settings (optional), and selecting EDIFACT identifiers to use for routing the outbound messages.

To configure the target message type:

. Expand the *Message type* section and click *Select*.
. In the *Select message type* window, do either of the following:

> * Select an existing send to message type and click *Select*.

Reviewer 3: ditto re. "send to"
I do think we need some text in this deliverable to explain UI terminology vs. actual conceptual terminology in narrative.

Reviewer 3: GLOBAL for "send to"

* Click *New* to create a new send to message type, as described in <<xref:partner-manager-create-message-type,Create and Configure Message Types>>.
. Click *Save*.
+
The message type and its validation settings are added to the message flow.
. Optionally configure acknowledgement settings for the outbound message.
. Optionally apply custom validation settings delimiters, and character encoding to the outbound message flow.
. In the *<protocol> Identifiers* section, select the sender and receiver identifiers to use on the X12 ISA and GS segments or the EDIFACT UNB segments.
+
These identifiers are configured in the host and partner profiles.

> === Configure Acknowledgment settings

Reviewer 3: ```suggestion
=== Configure Acknowledgment Settings
```

. In the *Acknowledgement* section, select *Expect functional acknowledgement* (X12) or *Expect CONTRL functional acknowledgment* (EDIFACT).
. Do either of the following:
* Select an existing endpoint at which to receive your partner's acknowledgement transaction.
* Click *New* to create a new endpoint, as described in <<xref:create-endpoint.adoc,Create and Configure Endpoints>>.
. In the *Mark overdue after* field, accept the default value of *24 Hours* or configure when you want the outbound transaction to be marked as overdue for an acknowledgement, according to the SLAs you have with your trading partner.

=== Optionally Apply Custom Validation Settings to EDIFACT and X12 Endpoints

By default, Partner Manager uses the EDIFACT or X12 send settings configured for the partner or 3rd party connection for message validation, delimiters, and character encoding.

To apply custom validation settings and delimiters to the outbound message flow, follow these steps:

. In the *EDIFACT Settings* section, select *Custom X12 settings* or *Custom EDIFACT settings* and provide custom configuration values.

> . In the *<message format> Settings* section

Reviewer 3: incomplete

[[send-to-partner-endpoint]]

> == Configure the Send To Partner Endpoint

Reviewer 3: ditto "send to" comments earlier

The send to partner endpoint is the target endpoint to which transformed messages are sent to partners, either directly or via 3rd party connections. Use a 3rd party-owned send to partner endpoint for partners that use a 3rd party connection.

Partner Manager supports the EDIFACT and X12 protocols for send to partner endpoints.

To create a send to partner endpoint:

. In the message flow, expand the *Send to <partner>* section and click *Select*.
. Do either of the following:
* Select an existing send to endpoint and click *Select*.
+

> You can select any previously configured Send to partner endpoints owned by the host, the partner for whom the message flow is configured, or a 3rd party connection.

Reviewer 3: note change in capitalization

+
* Click *New* to create a new send to endpoint, as described in <<xref:create-endpoint.adoc,Create and Configure Endpoints>>.

> If you selected AS2 as the protocol and there is no certificate associated with the partner or 3rd party profile, you must import the certificate of the partner or 3rd party connection that owns the endpoint.

Reviewer 3: ```suggestion
If you selected AS2 as the protocol and there is no certificate associated with the partner or third-party profile, import the certificate of the partner or third-party connection that owns the endpoint.
```

> == Verify That the Message Flow is Complete

Reviewer 3: ```suggestion
== Verify That the Message Flow Is Complete
```
verbs are always init. capped in titles

> Partner Manager dynamically validates the message flow configuration elements for completeness and displays a green checkmark if all the building blocks of the message flow are complete. After you verify the configurations, you are ready to deploy the message flow.

Reviewer 3: I would avoid the colloquial "building blocks"

> = Create an Outbound Message Flow

Reviewer 3: Edit gen. topic first
